src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseStream.cs

```diff
@@ -135,7 +135,11 @@ public override Task<int> ReadAsync(byte[] buffer, int offset, int count, Cancel
                 new TaskCompletionSource<int>(TaskCreationOptions.RunContinuationsAsynchronously);
             _state.TcsQueryDataAvailable.Task.ContinueWith((previousTask) => 
                 {
-                    if (previousTask.IsFaulted)
+                    if (token.IsCancellationRequested)
+                    {
+                        _state.TcsReadFromResponseStream.TrySetCanceled();
+                    }
```

Comment:
If this is necessary to do, it should be done after the subsequent else block that checks for faulted, as if the task failed, its failure should be propagated rather than ignoring the failure and marking it as canceled.

I also question whether this is necessary to check at all.  Are we not registering with the CancellationToken elsewhere to trigger cancellation of the operation?  If we are, then I'd expect previousTask to end as Canceled (barring race conditions, which we have even with such a check), and the subsequent previousTask.IsCanceled block to handle it.  But maybe we're not registering with the token?

Typically I recommend not checking for cancellation like this after an operation has already completed, as the results of the operation may be important to propagate (e.g. if the operation produced a disposable result).  In this case, maybe it doesn't matter, as the result is an integer, and you're likely using this as a place to cancel the overall composite operation that includes this one?

If this does need to stay, I'd suggest changing ```TrySetCanceled()``` to ```TrySetCanceled(token)```.